virttest/qemu_vm.py

```diff
@@ -1100,6 +1100,7 @@ def add_memorys(devices, params):
                 if params.get("hugepage_path"):
                     backend_options["backend_mem"] = "memory-backend-file"
                     backend_options["mem-path_mem"] = params["hugepage_path"]
+                    backend_options["prealloc_mem"] = params.get("vm_prealloc_mem", "yes")
```

Comment:
As what @PaulYuuu pointed out, we'd better support setting the `prealloc` option for non-hugepage case as well, within this context.

Reply:
@luckyh It's not needed, only in a few already-implemented cases prealloc is used for others than `memory-backend-file`

Reply:
Even though we have test cases covering prealloc, the VM default memory backend also supports all attrs.

My concern is this: in the future, we want to support another option, add support to VT, then support another option, and then add support again, this cycle continues.

The history of vm_mem is at the first, only support `memory-backend-ram`, and then supports custom `vm_mem_backend` and hugepage, and now we want to support a new option.

Reply:
@mcasquer I understand the original intention of this patch.
But other backends that the maintainer considers for support also need to be considered.
Because the current framework does not support setting preallocate for all memory-backends.
For example, if we want to test `prealloc` on `memory-backend-memfd` in the future, then we need to issue a similar patch to support it.
So it makes sense to support it in all memory backends this time.

Reply:
@PaulYuuu @zhenyzha I got the idea now. In that case we can do it now for all the memory-backend options.

Reply:
I guess the `backend_options["prealloc_mem"]` has to be extracted from the `hugepage_path` condition